src/item.cpp

```diff
@@ -2965,13 +2965,29 @@ units::volume item::get_storage() const
 }
 
 int item::get_env_resist() const
+{
+    const auto t = find_armor_data();
+    if( t == nullptr ) {
+        return 0;
+    }
+    // modify if item is a gas mask and has filter
+    int resist_base = static_cast<int>( static_cast<unsigned int>( t->env_resist ) );
+    int resist_filter = get_var( "overwrite_env_resist", 0 );
+    int resist = std::max( resist_base, resist_filter );
+    // modify based on item health
+    float eff_resist = max_damage() + 1 - damage();
+    eff_resist /= max_damage() + 1;
```

Comment:
You've just declared variable `eff_resist` with some value and immediately on the next step you are changing it. If I'm not mistaken, you can inline it in one step.

Reply:
Single line `float eff_resist = ( max_damage() + 1 - damage() ) / ( max_damage() + 1 ) ;` not working the way I think it should and provide zero resistance for any damage. Looks like it is calculated in the wrong order.

Reply:
`damage()` returns `int`, so does `max_damage()`, and `1` is also a `int`. The sum of them is also a `int`. `int / int` does integer division. You need to cast one of the two operands to the division to `float`.

Reply:
Now I see the problem, thanks. So new function get_relative_health().